File: frontend/src/domain/Job.ts
// frontend/src/domain/Job.ts
export interface TriggerAccepted {
  job_id: string
  message: string
}

export interface JobStatus {
  job_id: string
  status: 'running' | 'done' | 'error'
  report_id: number | null
  error: string | null
}

export interface TriggerParams {
  start_date?: string
  end_date?: string
}

File: backend/src/domain/entities/job.py
# backend/src/domain/entities/job.py
from dataclasses import dataclass
from enum import StrEnum
from typing import Optional


class JobStatus(StrEnum):
    PENDING = "pending"
    RUNNING = "running"
    DONE    = "done"
    ERROR   = "error"


@dataclass(frozen=True)
class JobRecord:
    job_id: str
    status: JobStatus
    report_id: Optional[int] = None
    error: Optional[str] = None
